src/modules/ethereum/chains/ChainConfigStore.ts

```diff
@@ -22,36 +17,65 @@ function remove<T>(arr: T[], predicate: (item: T) => boolean) {
   }
 }
 
-export class ChainConfigStore extends PersistentStore<ChainConfig> {
-  static initialState: ChainConfig = { ethereumChains: [] };
+class ChainConfigStore extends PersistentStore<ChainConfig> {
+  static initialState: ChainConfig = {
+    version: 2,
+    ethereumChainConfigs: [],
+  };
+
+  async restore() {
+    const saved = await PersistentStore.readSavedState<ChainConfig>(this.key);
+    if (saved) {
+      this.setState(upgradeRecord(saved, upgrades));
+    }
+    this.isReady = true;
+    this.checkChainsForUpdates();
+  }
+
+  readyPromise = this.restore();
```

Comment:
Why overwrite this? The same code is being called in `PersistentStore`'s constructor